Raven.Database/Bundles/MoreLikeThis/MoreLikeThis.cs

```diff
@@ -802,7 +803,20 @@ internal PriorityQueue<object[]> RetrieveTerms(RavenJObject jsonDocument)
 
                 if (value == null) continue;
 
-                AddTermFrequencies(new StringReader(value.ToString()), words, fieldName);
+                switch (value.Type)
+                {
+                    case JTokenType.Array:
+                        foreach (var item in (RavenJArray)value)
+                        {
+                            if (item.Type != JTokenType.String) break;
+                            AddTermFrequencies(new StringReader(item.Value<string>()), words, fieldName);
+                        }
+                        break;
+                    case JTokenType.String:
+                        AddTermFrequencies(new StringReader(value.Value<string>()), words, fieldName);
+                        break;
+                    default: continue;
```

Comment:
:+1: 

Done and added a [new test](https://github.com/ayende/ravendb/pull/4048/files#diff-7da0aa718ef27b483ceb03ca4c87b0bcR140) to assert that throws for non-supported types.